src/os/ui/draw/basedrawcontrols.js

```diff
@@ -161,6 +164,7 @@ os.ui.draw.BaseDrawControlsCtrl.prototype.disposeInternal = function() {
   os.dispatcher.unlisten(os.ui.draw.DrawEventType.DRAWPOLYGON, this.onDrawType, false, this);
   os.dispatcher.unlisten(os.ui.draw.DrawEventType.DRAWLINE, this.onDrawType, false, this);
   os.dispatcher.unlisten(os.ui.draw.DrawEventType.DRAWCOUNTRY, this.onDrawType, false, this);
+  os.dispatcher.unlisten(os.ui.draw.DrawEventType.DRAWWATER, this.onDrawType, false, this);
```

Comment:
As part of the refactor I mention below, things like these enumerated event types should not be referenced outside of the classes that create them.